cdap-watchdog/src/main/java/co/cask/cdap/logging/gateway/handlers/AbstractChunkedLogProducer.java

```diff
@@ -65,6 +65,9 @@ public ByteBuf nextChunk() throws Exception {
     if (!hasStarted) {
       hasStarted = true;
       startBuffer = Unpooled.copiedBuffer(onWriteStart());
+      if (startBuffer.isReadable()) {
```

Comment:
I'm not sure I understand how this fixes the bug. Isn't it still the case that the end of the json array is not written?

Reply:
Instead of fixing in here, I think it's cleaner to fix the `hasFinished` condition below.

```java
if (!hasFinished) {
  hasFinished = true;
  return Unpooled.wrappedBuffer(startBuffer, onWriteFinish());
}
```

Reply:
ok I think I understand what's happening now. +1 to terence's suggestion.